File: 3-ordenacao/Java/Principal.java
import java.util.ArrayList;

import javax.swing.JOptionPane;
import java.util.Collections;

public class Principal {

    public static void main(String a[]) {
        ArrayList<Integer> listaBolha = new ArrayList<Integer>();
        ArrayList<Integer> listaSelecao = new ArrayList<Integer>();
        ArrayList<Integer> listaPente = new ArrayList<Integer>();
        long tempoInicio, tempoFim;  
        
        int quantidade = Integer.parseInt(JOptionPane.showInputDialog(null,"Quantos números quer gerar? "));

        Util.popular(quantidade, listaBolha);
        listaSelecao.addAll(listaBolha);
        listaPente.addAll(listaBolha);

        

        // //bolha
        // tempoInicio = System.nanoTime();
        // MetodosOrdenacao.bolha(listaBolha);
        // tempoFim = System.nanoTime();
        // System.out.println("Tempo (ms) bolha: " + (tempoFim - tempoInicio)/1000000);

        // //selecao
        // tempoInicio = System.nanoTime();
        // MetodosOrdenacao.selecao(listaSelecao);
        // tempoFim = System.nanoTime();
        // System.out.println("Tempo (ms) seleção: " + (tempoFim - tempoInicio)/1000000);


        
        //pente
        tempoInicio = System.nanoTime();
        MetodosOrdenacao.pente(listaPente);
        tempoFim = System.nanoTime();
        System.out.println("Tempo (ms) pente: " + (tempoFim - tempoInicio)/1000000);
        // Util.exibir(listaPente);
               
        System.exit(0);
    }
}

File: 3-ordenacao/CPP/principal.cpp
#include <iostream>
#include <time.h>

#include "metodosOrdenacao.h"
#include "utilidades.h"

using namespace std;

int main() {
    clock_t tempoInicio, tempoFim;
    long long int tamanho;

    system("clear");//no windows é system("cls");

    cout << "Tamanho do vetor: ";
    cin >> tamanho;
    int *vetorBolha;
    int *vetorSelecao;
    vetorBolha = (int*)malloc(sizeof(int) * tamanho);
    vetorSelecao = (int*)malloc(sizeof(int) * tamanho);
    popular(vetorBolha, tamanho);
    copiar(vetorBolha, tamanho, vetorSelecao);
    //exibir(vetorSelecao, tamanho);

    //bolha
    tempoInicio = clock();
    bolha(vetorBolha,tamanho);
    tempoFim = clock();
    cout << "Tempo para bolha: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
    // exibir(vetorBolha, tamanho);

    //selecao
    tempoInicio = clock();
    selecao(vetorSelecao,tamanho);
    tempoFim = clock();
    cout << "Tempo para selecao: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
    //exibir(vetorSelecao, tamanho);

    return 1;
}